docs/dev_guide/custom_field/ajax_custom_formatter.rst

```diff
@@ -0,0 +1,105 @@
+AJAX Responsive Formatters
+===========================
+
+
+Some fields need to be responsive.  For example, a user might select an analysis or organism to display data from. Drupal developers often use AJAX to rebuild the page based on user input.
+
+Drupal and AJAX
+---------------
+
+Drupal has its special way of doing AJAX!  You should `read the documentation carefully! <https://api.drupal.org/api/drupal/includes%21ajax.inc/group/ajax/7.x>`_ To Drupal, AJAX only makes sense as on forms, and field formatters **are not forms**.  Instead, they are `renderable arrays <https://www.drupal.org/docs/7/api/render-arrays/render-arrays-overview>`_
```

Comment:
```suggestion
Drupal has its own special way of doing AJAX! This is important to ensure that changes are executed in the correct order. You should `read the documentation carefully! <https://api.drupal.org/api/drupal/includes%21ajax.inc/group/ajax/7.x>`_  The Drupal AJAX API works best on forms, and field formatters **are not forms**.  Instead, they are `renderable arrays. <https://www.drupal.org/docs/7/api/render-arrays/render-arrays-overview>`_
```

Reply:
>This is important to ensure that changes are executed in the correct order. 

I dont actually understand what this sentence is saying...

Reply:
>  The Drupal AJAX API works best on forms

You imply that the Drupal way of handling AJAX is do-able on renderable arrays, but that its better on forms.  So, can we do it on renderable arrays *without forms* as this sentence implies?  I didn't think so, thats why we have to use a form instead.

Reply:
No, drupal AJAX can't be done on renderable arrays but it also doesn't only work on forms -that's just the most common way. See https://www.drupal.org/docs/7/api/javascript-api/simple-drupal-ajax-load-with-jquery-and-delivery-callback.